pulp_ansible/app/migrations/0030_collectionversion_requires_ansible.py

```diff
@@ -0,0 +1,43 @@
+# Generated by Django 2.2.17 on 2020-12-09 20:22
+
+import json
+import tarfile
+import yaml
+from django.db import migrations, models
+
+from pulp_ansible.app.tasks.utils import get_file_obj_from_tarball
+
+
+def set_requires_ansible_and_manifest_and_files_json(apps, schema_editor):
+    CollectionVersion = apps.get_model("ansible", "CollectionVersion")
+    for collection_version in CollectionVersion.objects.all():
+        artifact = collection_version.contentartifact_set.get().artifact
+        with artifact.file.open() as artifact_file, tarfile.open(fileobj=artifact_file, mode="r") as tar:
+            runtime_metadata = get_file_obj_from_tarball(
+                tar, "meta/runtime.yml", artifact.file.name, raise_exc=False
+            )
+            if runtime_metadata:
+                runtime_yaml = yaml.safe_load(runtime_metadata)
+                collection_version.requires_ansible = runtime_yaml.get("requires_ansible")
+            collection_version.manifest = json.load(
+                get_file_obj_from_tarball(tar, "MANIFEST.json", artifact.file.name)
+            )
+            collection_version.files = json.load(get_file_obj_from_tarball(tar, "FILES.json", artifact.file.name))
+            collection_version.save()
```

Comment:
One idea I had was to wrap this whole section in a general try/except blog that logs and continues? I'm not sure if that's a great idea, but it would be better to continue if a fatal exception was experienced due to one or a small number of data quality issues (e.b. a missing FILES.json, invalid yaml, missing MANIFEST.json). It would make this migration more "defensive". Wdyt about this?

Reply:
I think we can address it by passing `raise_exc=False`

Reply:
We can but what about the invalid yaml case?

Reply:
Oh and also I think we want this to be logged to the user, which that wouldn't do.